hail/src/main/scala/is/hail/expr/ir/Env.scala

```diff
@@ -129,6 +137,29 @@ case class BindingEnv[V](
     newEnv
   }
 
+  def subtract[T](bindings: Bindings[T]): BindingEnv[V] = {
+    val Bindings(eval, agg, scan, relational, _) = bindings
+    var newEnv = modifyWithoutNewBindings(bindings)
+    agg match {
+      case AggEnv.Create(bindings) =>
+        newEnv = newEnv.copy(agg = Some(newEnv.agg.get.delete(bindings.view.map(_._1))))
```

Comment:
This is a feature from the old implementation that I utterly despise - `None.get` is a smell that gives terrible error messages. Somewhere in your stack of PRs can you think of a way to remove this? I wont make a massive thing of this is probably a side-quest to a side-quest, but this sort of thing just makes me sad.

Reply:
I hear you, but I do think it would be a tangent to find a clean way to improve the error messages everywhere we do this, when all this work is pushing towards killing the agg/scan environments entirely, which will make this a non-issue.